src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -557,9 +757,50 @@ def _get_env_and_group_from_log_analytics(
                 env_list = [e for e in env_list if e["location"] == location]
             if env_list:
                 # TODO check how many CA in env
-                env_details = parse_resource_id(env_list[0]["id"])
-                env.set_name(env_details["name"])
-                resource_group.name = env_details["resource_group"]
+                # set_name() with parse name and resource_group from environment id
+                env.set_name(env_list[0]["id"])
+                resource_group.name = env.resource_group.name
+
+
+def _get_custom_location_and_extension_id_and_location_from_cluster(
+        cmd,
+        env: "ContainerAppEnvironment",
+        custom_location: "CustomLocation",
+        extension: "Extension"
+):
+    #  if connected cluster have one custom location (with ext namespace) bind to the container app ext, then use it.
+    if custom_location.name is None and custom_location.connected_cluster_id is None:
+        raise ValidationError(
+            "please specify one of connected-cluster-id or custom location you want to create the Connected environment.")
+
+    if custom_location.connected_cluster_id and env.custom_location_id is None:
```

Comment:
If the command specified location and connected cluster Id, but the connected cluster is not in the same location with the specified location. what's the location of customlocation?

Reply:
The locations of custom location and connected cluster need to be consistent.
The location of custom location get from connected cluster.